lib/render/error.php

```diff
@@ -26,6 +26,7 @@ function RenderErrorCodeWarning(?string $errorCode)
         'issue_submitted' => "Your issue ticket has been successfully submitted.",
         'merge_failed' => "Problems encountered while performing merge. These errors have been reported and will be fixed soon... sorry!",
         'merge_success' => "Game merge successful!",
+        'messageblocked' => "Message was not delivered - blocked by recipient.",
```

Comment:
Not sure of the implications. Do we want to expose that a user blocked another?

Reply:
I'm not sure what the correct behavior here is. It seems the most common implementation is to pretend it worked, without actually doing anything.

Reply:
Yeah, that's my thought, too. Otherwise it might be seen as a privacy issue (I would).
I'd say we have a a look at other platforms, how they deal with blocked messages.

Reply:
From a brief trip through Google search results, it seems like there's really only two common implementations:
1) All activity related to the blocking user is removed from the blocked user's visibility.
    - As far as the blocked user knows, the blocking user no longer exists.
2) The blocked user has no way of knowing they've been blocked. 
    - The blocking user still exists, but does not receive any direct communication from the blocked user. 
    - The blocked user can only assume they're being ignored by the blocking user, but in reality they're being ignored by the site.
- In some cases, the blocked user is also scrubbed from the blocking user's visibility. Either fully, or with spoiler-like tags so the blocking user can choose to see the blocked user's content.

The provided implementation doesn't really fit into either of those.

The ability to directly message the user was explicitly removed, making it obvious that they've been blocked. The ability to post on the blocking user's wall was also removed, but the blocked user could attribute that to the user disabling wall comments.

The highlighted point of contention comes from the Inbox implementation. Once you reply to any Inbox message, you can change the recipient (it even updates the user's image when doing so!), so you could reply to another user and still target the blocked user. This change generates an error at the last possible moment in case any other workarounds exist.

I'll change the code to return `sentok` in this situation.

As the "Send Private Message" directs to the same page as the Reply functionality, it could be handled in the same way to hide the fact that the user has been blocked. The only remaining indication would be the lack of an "Add Friend" button.

Reply:
Thanks for going in-depth with this.

> I'll change the code to return `sentok` in this situation.

Seems to be the most sensible solution for now, I concur.

> The only remaining indication would be the lack of an "Add Friend" button.

Still an indicator but i guess that's fine for now, too.
It's mainly the `messageblocked` hint that I'd want to avoid - the reasons for not delivering a message might be varying - exposing this one is very explicit.